### PR TITLE
feat(core): support async-node target

### PR DESCRIPTION
**Pull request by hardfist**
## Summary
support `async-node` target and enable webpack-tests for async-node target, hmr may have problem but I can't test it now, because the test suits not support async-node target well yet

### <samp>🤖 Generated by Copilot at 8764410</samp>

No summary available (Limit exceeded: required to process 136090 tokens, but only 50000 are allowed per call)

<details open=true>
  <summary><h2>Walkthrough</h2></summary>

### <samp>🤖 Generated by Copilot at 8764410</samp>

No walkthrough available (Limit exceeded: required to process 136090 tokens, but only 50000 are allowed per call)

</details>

### CODE REVIEW

File: packages/rspack/src/config/zod/target.ts [hunk 1]
```diff
@@ -17,6 +17,7 @@ const allowTarget = z
 		"browserslist"
 	])
 	.or(z.literal("node"))
+	.or(z.literal("async-node"))
 	.or(
 		z.custom<`node${number}`>(
 			value => typeof value === "string" && /^node\d+$/.test(value)
```
Comment (hyf0) [inline packages/rspack/src/config/zod/target.ts:23]: @ahabhgk Does `async-node` support pattern like `async-node${number}`?

File: crates/rspack_plugin_runtime/src/runtime_module/runtime/readfile_chunk_loading.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+var installChunk = (chunk) => {
```
Comment (underfin) [inline crates/rspack_plugin_runtime/src/runtime_module/runtime/readfile_chunk_loading.js:1]: Here should be es5

File: crates/rspack_plugin_runtime/src/runtime_module/readfile_chunk_loading.rs [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+          .replace("JS_MATCHER", "chunkId")
+          .replace("$OUTPUT_DIR$", &root_output_dir),
+      ));
+    }
+    if with_hmr {
+      source.add(RawSource::from(include_str!(
+        "runtime/readfile_chunk_loading_with_hmr.js"
+      )));
+      source.add(RawSource::from(
+        include_str!("runtime/javascript_hot_module_replacement.js").replace("$key$", "jsonp"),
```
Comment (underfin) [inline crates/rspack_plugin_runtime/src/runtime_module/readfile_chunk_loading.rs:105]: `jsonp` should be `readFileVm`.

File: crates/rspack_plugin_runtime/src/runtime_module/runtime/readfile_chunk_loading.js (hunk 2 above)
Comment (hardfist) [reply crates/rspack_plugin_runtime/src/runtime_module/runtime/readfile_chunk_loading.js:1]: does node need es5?

**Comment by ahabhgk** [comment]
!eco-ci

**Review by underfin — APPROVED** [review]

**Pull request merged**